### PR TITLE
feat(button): align with 2018 material design spec

### PR DESCRIPTION
**Pull request by crisbeto**
Aligns the button component with the latest Material design spec. 
before
after

### CODE REVIEW

File: src/lib/button/_button-base.scss [hunk 1]
```diff
@@ -44,15 +39,16 @@ $mat-mini-fab-padding: 8px !default;
...
   text-align: center;
 
   // Sizing.
   margin: $mat-button-margin;
   min-width: $mat-button-min-width;
-  line-height: $mat-button-line-height;
+  height: $mat-button-height;
   padding: $mat-button-padding;
   border-radius: $mat-button-border-radius;
+  letter-spacing: 0.08929em;
```
Comment (josephperrott) [inline src/lib/button/_button-base.scss]: Letter spacing should be handled in the typography correct?

Comment (crisbeto) [reply src/lib/button/_button-base.scss]: At least at the moment we don't have letter spacing as a part of the config. I can add it in when I get to doing the typography changes.

**Comment by jelbourn** [comment]
I'm iffy about making fabs not default to accent

**Comment by josephperrott** [comment]
@jelbourn Is your concern with the amount of change this will bring?  Or a bad default state?

**Comment by jelbourn** [comment]
@josephperrott mainly that it's _too_ breaking of a change, but also that I do suspect most people won't want a default (gray) FAB.

**Comment by crisbeto** [comment]
I've moved out the letter spacing into the typography mixin and will hold off on switching the default color for FABs @josephperrott.

**Comment by josephperrott** [comment]
@jelbourn okay, works for me.  Lets leave it as the default for now and we can look in the future at allowing for a provider to defines the default for an app.

**Comment by josephperrott** [comment]
Removing merge ready as we plan to rework this PR a bit.

**Comment by crisbeto** [comment]
I've reworked the PR as discussed. The screenshot above has been updated as well. Mark as merge ready again.

**Pull request merged**